docs/migration-guide.md

```diff
@@ -58,171 +82,63 @@ To add Storybook to a project that isn’t currently using Storybook:
-3. Search [Storybook issues on GitHub](https://github.com/storybookjs/storybook/issues). If you’re seeing a problem, there’s a good chance other people are too. If so, upvote the issue, try out any workarounds described in the comments, and comment back if you have useful info to contribute.
-4. If there’s no existing issue, you can [file one](https://github.com/storybookjs/storybook/issues/new/choose), ideally with a reproduction attached. We’ll be on top of Storybook 7 issues as we’re stabilizing the release.
-
-If you prefer to debug yourself, here are a few useful things you can do to help narrow down the problem:
-
-1. Try removing all addons that are not in the `@storybook` npm namespace. Community addons that work well with 6.x might not yet be compatible with 7.0, and this is the fastest way to isolate that possibility. If you find an addon that needs to be upgraded to work with Storybook 7, please post an issue on the addon’s repository, or better yet, a PR to upgrade it!
-2. Another debugging technique is to bisect to older prerelease versions of Storybook to figure out which release broke your Storybook. For example, assuming that the current prerelease of Storybook is `7.0.0-beta.56`, you could set the version to `7.0.0-alpha.0` in your `package.json` and reinstall to verify that it still works (alpha.0 should be nearly identical to `6.5.x`). If it works, you could then try `7.0.0-beta.0`, then `7.0.0-beta.28` and so forth. Once you’ve isolated the bad release, read through its [CHANGELOG](https://github.com/storybookjs/storybook/blob/next/CHANGELOG.md) entry and perhaps there’s a change that jumps out as the culprit. If you find the problem, please submit an issue or PR to the Storybook repo and we’ll do our best to take care of it quickly.
-
-## Optional migrations
-
-In addition to the automigrations and manual migrations above, there are also optional migrations that you should consider. These are things that we’ve deprecated in Storybook 7 (but remain backwards compatible), or best practices that should help you be more productive in the future.
-
-These include:
-
-- Upgrade to Component Story Format v3 (CSF3), the latest iteration of our example format
-- Upgrade from the legacy storiesOf API to Component Story Format
-- Upgrade from `.stories.mdx` files to pure MDX and Component Story Format
-- Upgrade from Storybook 6 recipe “CSF stories with arbitrary MDX”
-
-### CSF2 to CSF3
-
-We published a [detailed post about CSF3](https://storybook.js.org/blog/storybook-csf3-is-here/) including all of its benefits and migration steps.
-
-If you want to just skip to the migration, we provide a codemod for your convenience which should automatically make the code changes for you (make sure to update the glob to fit your files):
+Then, you can optionally convert CSF 2 to CSF 3 ([why should you do this?](https://storybook.js.org/blog/storybook-csf3-is-here/)):
```

Comment:
Note that `storiesOf-to-CSF` converts to CSF1. There is no automated transformation from CSF 1 to 2.

Reply:
@shilman — Here's what I came up with:

> ```sh
> # Convert storiesOf to CSF 1
> npx storybook@latest migrate storiesof-to-csf --glob="**/*.stories.tsx" --parser=tsx
> ```
> 
> While stories written in CSF 1 are supported in Storybook 8, we recommend upgrading to CSF 3, which has [significant benefits](https://storybook.js.org//blog/component-story-format-3-0/) when writing and maintaining stories. You'll need to manually [migrate your stories to CSF 2](https://storybook.js.org//blog/introducing-storybook-args/), and then you can use the [codemod to automatically upgrade from CSF 2 to 3](https://github.com/storybookjs/storybook/blob/docs-migration-8/docs/migration-guide.md#csf-2-to-csf-3).

I don't love it. 😕 But we don't have any materials that go straight from CSF 1 to 3, and providing a true guide for that (step-by-step, with multiple renderer snippets) would be a ton of effort for a small set of users that will encounter this. Wdyt?

Reply:
@kylegach I don't know if there's much value in converting from 1 to 2 and then 2 to 3.

How about something like this:

```
# Convert storiesOf to CSF 1
npx storybook@latest migrate storiesof-to-csf --glob="**/*.stories.tsx" --parser=tsx
```

This will create your stories to CSF1 stories, which are story functions that don't accept args. These are fully supported in Storybook 8, and will continue to be for the foreseeable future. However, we recommend writing all new stories in CSF3, which are story objects that are easier to write, reuse, and maintain.

Reply:
My thinking was that CSF 1 (no args) to CSF 3 (implicit render functions, stories are objects, etc) would be quite a leap. So easier for the manual part to go to CSF 2, then let the automod handle it.

But I agree that it's weird.

-----

Oh, I just caught the "all _new_ stories" bit of your edit! That works for me. Thanks!